### ISSUE TITLE
Further question about trainable and trainning

### ISSUE THREAD

**Issue by weizhenzhao** [CyberZHG/keras-bert#158]
Hi   @CyberZHG 

![image](https://user-images.githubusercontent.com/13706689/67155639-a3ecaa00-f345-11e9-9c3d-37f3fcf5944b.png)
![image](https://user-images.githubusercontent.com/13706689/67155646-cda5d100-f345-11e9-94a8-7f01abd137f3.png)
I ' m not very clear about the trainning and trainable
I set the output_num = 4 , it means concat the last 4 layers' output from the pretrain bert model
and I added a lstm layer with GlobalMax pool on top of that.
if I also want to train the variables in the   ( lstm + maxpool + last 4 layers' )
how should I set the training and trainable  value?

Thanks
weizhen

**Comment by CyberZHG** [comment]
```python
training=False,
output_num=4,
trainable=['Encoder-9', 'Encoder-10', 'Encoder-11', 'Encoder-12'],
```

**Comment by weizhenzhao** [comment]
Hi @CyberZHG

how to get the embedding of '[cls]' in keras-bert?

I want to concat the '[cls]' character embedding with the last four layers's embedding?
than append self-attention and globalmax pool on top of that?
`import tensorflow as tf
import keras as keras
import json
import keras.backend as K
from keras.models import load_model
from keras.utils import to_categorical
from keras_bert import load_trained_model_from_checkpoint, load_vocabulary, get_model
from keras_bert import Tokenizer, loader
from keras_bert import AdamWarmup, calc_train_steps

import pandas as pd
import numpy as np
from tqdm import tqdm
import gc
import os
import csv
# from keras_radam import RAdam
from keras_bert.layers import MaskedGlobalMaxPool1D
from keras_self_attention import SeqSelfAttention

SEQ_LEN = 256
BATCH_SIZE = 15
EPOCHS = 2
LR = 2e-5
layer_num = 2

pretrained_path = './chinese_roberta_wwm_ext_L-12_H-768_A-12'
config_path = os.path.join(pretrained_path, 'bert_config.json')
checkpoint_path = os.path.join(pretrained_path, 'model.ckpt-3572')
vocab_path = os.path.join(pretrained_path, 'vocab.txt')

DATA_COLUMN = 'comment_text'
LABEL_COLUMN = 'target'

token_dict = load_vocabulary(vocab_path)
tokenizer = Tokenizer(token_dict)


def read_csv(input_file, quotechar=None):
    with tf.gfile.Open(input_file, "r") as f:
        reader = csv.reader(f, delimiter="\t")
        lines = []
        for line in reader:
            lines.append(line)
        return lines


def process_data():
    lines = read_csv('./data/finance_train.csv')
    indices, targets = [], []
    for (i, line) in enumerate(lines):
        split_line = line[0].strip().split('*')
        id = split_line[3]
        text_a = split_line[1]
        text_b = split_line[2]
        label = split_line[0]
        if "negative" == label:
            lab = 1
        else:
            lab = 0
        ids, segments = tokenizer.encode(text_a, second=text_b, max_len=SEQ_LEN)
        indices.append(ids)
        targets.append(lab)
    items = list(zip(indices, targets))
    np.random.shuffle(items)
    indices, targets = zip(*items)
    indices = np.array(indices)
    return [indices, np.zeros_like(indices)], targets


def process_test():
    lines = read_csv('./data/finance_test.csv')
    indices, targets = [], []
    id_label = []
    entity_list = []
    for (i, line) in enumerate(lines):
        split_line = line[0].strip().split('*')
        id = split_line[3]
        text_a = split_line[1]
        text_b = split_line[2]
        label = split_line[0]
        ids, segments = tokenizer.encode(text_a, second=text_b, max_len=SEQ_LEN)
        indices.append(ids)
        if "negative" == label:
            lab = 1
        else:
            lab = 0
        targets.append(lab)
        id_label.append(id)
        entity_list.append(text_b)
    # items = list(zip(indices, targets))
    # np.random.shuffle(items)
    # indices, targets = zip(*items)
    indices = np.array(indices)
    return [indices, np.zeros_like(indices)], id_label, entity_list, targets


data_x, data_y = process_data()
data_y = to_categorical(data_y)
test_x, id_label, entity_list, test_targets = process_test()
gc.collect()

# model = load_trained_model_from_checkpoint(
#     config_path,
#     checkpoint_path,
#     seq_len=SEQ_LEN,
# )

with open(config_path, 'r') as reader:
    config = json.loads(reader.read())

# model = load_trained_model_from_checkpoint(
#     config_path,
#     checkpoint_path,
#     training=True,
#     trainable=True,
#     seq_len=SEQ_LEN,
# )
#
# inputs = model.inputs[:2]
# dense = model.get_layer('NSP-Dense').output
# outputs = keras.layers.Dense(units=2, activation='softmax')(dense)
#
# model = keras.models.Model(inputs, outputs)
# model.compile(
#     RAdam(lr=LR),
#     loss='sparse_categorical_crossentropy',
#     metrics=['sparse_categorical_accuracy'],
# )
#
# import tensorflow as tf
# import tensorflow.keras.backend as K
#
# sess = K.get_session()
# uninitialized_variables = set([i.decode('ascii') for i in sess.run(tf.report_uninitialized_variables())])
# init_op = tf.variables_initializer(
#     [v for v in tf.global_variables() if v.name.split(':')[0] in uninitialized_variables]
# )
# sess.run(init_op)
#
# model.fit(
#     data_x,
#     data_y,
#     epochs=EPOCHS,
#     batch_size=BATCH_SIZE,
#     validation_split=0.3
# )
model = get_model(
    token_num=config['vocab_size'],
    pos_num=config['max_position_embeddings'],
    seq_len=SEQ_LEN,
    embed_dim=config['hidden_size'],
    transformer_num=config['num_hidden_layers'],
    head_num=config['num_attention_heads'],
    feed_forward_dim=config['intermediate_size'],
    feed_forward_activation=config['hidden_act'],
    training=False,
    trainable=None,
    #trainable=['Encoder-9', 'Encoder-10', 'Encoder-11', 'Encoder-12'],
    # trainable=['Encoder-{}-MultiHeadSelfAttention-Adapter'.format(12 - i) for i in range(layer_num)] +
    #           ['Encoder-{}-FeedForward-Adapter'.format(12 - i) for i in range(layer_num)] +
    #           ['Encoder-{}-MultiHeadSelfAttention-Norm'.format(12 - i) for i in range(layer_num)] +
    #           ['Encoder-{}-FeedForward-Norm'.format(12 - i) for i in range(layer_num)],
    output_layer_num=layer_num)

inputs, outputs = model
model1 = keras.models.Model(inputs=inputs, outputs=outputs)
loader.load_model_weights_from_checkpoint(model1, config, checkpoint_path, training=False)

total_steps, warmup_steps = calc_train_steps(
    num_example=data_x[0].shape[0],
    batch_size=BATCH_SIZE,
    epochs=EPOCHS,
    warmup_proportion=0.1,
)

optimizer = AdamWarmup(total_steps, warmup_steps, lr=2e-5, min_lr=2e-6)


def get_custermized_model(model):
    # lstm = keras.layers.Bidirectional(keras.layers.LSTM(100, return_sequences=True,
    #                                                     dropout=0.1,
    #                                                     recurrent_dropout=0.1))(model.output)
    att = SeqSelfAttention(attention_type=SeqSelfAttention.ATTENTION_TYPE_MUL,
                           kernel_regularizer=keras.regularizers.l2(1e-4),
                           bias_regularizer=keras.regularizers.l1(1e-4),
                           attention_regularizer_weight=1e-4,
                           name='Attention')(model.output)
    pool_layer = MaskedGlobalMaxPool1D(name='Pooling')(att)
    x = keras.layers.Dense(units=2, activation='softmax')(pool_layer)
    custermized_model = keras.models.Model(inputs=model.inputs, outputs=x)
    custermized_model.compile(loss='binary_crossentropy', optimizer=optimizer, metrics=['acc'])
    return custermized_model


custermized_model = get_custermized_model(model=model1)
history = custermized_model.fit(x=data_x, y=data_y, epochs=EPOCHS)

predicts = custermized_model.predict(test_x)
predict_label = np.argmax(predicts, axis=1)
print("test_targets")
print("len test_targets")
print(len(test_targets))
print(test_targets)

print("predict_targets")
print("len predict_targets")
print(len(predict_label))
print(predict_label)
correct_count = 0
for i in range(len(predict_label)):
    if test_targets[i] == predict_label[i]:
        correct_count = correct_count + 1

print(correct_count / (len(predict_label) * 1.0))

predicts = custermized_model.predict(test_x)
with open("./result.csv", "w") as fout:
    for i in range(len(id_label)):
        fout.write(str(id_label[i]))
        fout.write(',')
        fout.write(str(predicts[i][0]))
        fout.write(',')
        fout.write(str(predicts[i][1]))
        fout.write(',')
        fout.write(entity_list[i])
        fout.write('\n')
`

Thanks
weizhen
